source/install/install-debian-88-mattermost.rst

```diff
@@ -74,8 +74,9 @@ Assume that the IP address of this server is 10.10.10.2.
     Requires=postgresql.service
 
     [Service]
-    Type=simple
+    Type=notify
     ExecStart=/opt/mattermost/bin/platform
+    TimeoutStartSec=90

```

Comment:
NB: this is the start timeout. If the server doesn't send the ready notification within 90s, systemd will consider it as stuck (which makes sense, for instance if the server is waiting for an unavailable DB connection).

90s is the default system-wide value provided by systemd. I prefer to make it explicit in this file, to make it clear that the server taking too long to start could result in the process being killed.

That said, it means that if for instance a database migration is performed on the main thread before the server is signaled as ready, it should take no more of 90s, otherwise the process will be flagged as unresponsive and killed. 

Do database migrations happen in the background, or synchronously on startup? And if so, what is the expected maximum duration of these migrations?

Reply:
Synchronously on startup. The duration totally depends on the size of the deployment, but 90 seconds is definitely within the realm of possibility. I would suggest something a little more extreme to be safe, like 3600.